apps/hash-graph/lib/graph/src/ontology.rs

```diff
@@ -157,9 +177,26 @@ impl OntologyType for EntityType {
         self.id()
     }
 
-    #[cfg(feature = "type-fetcher")]
     fn traverse_references(&self) -> Vec<OntologyTypeReference> {
-        traverse_entity_type_references(self).collect()
+        self.property_type_references()
+            .into_iter()
+            .map(OntologyTypeReference::PropertyTypeReference)
+            .chain(
+                self.inherits_from()
+                    .all_of()
+                    .iter()
+                    .map(OntologyTypeReference::EntityTypeReference),
+            )
+            .chain(self.link_mappings().into_iter().flat_map(
+                |(link_entity_type, destination_entity_type_constraint)| {
+                    {
+                        once(link_entity_type)
+                            .chain(destination_entity_type_constraint.unwrap_or_default())
+                    }
+                    .map(OntologyTypeReference::EntityTypeReference)
+                },
+            ))
```

Comment:
This implementation avoids the extra allocation we did before. `<&[]>::default()` returns an empty slice

Reply:
`destination_entity_type_constraint` is of type `Option<&[]>`. Calling `.unwrap_or_default` will either return the slice or an empty slice.

Reply:
Previously we used a temporary vector